capstone/src/main/webapp/script.js

```diff
@@ -110,25 +130,211 @@ async function getAnalysis(text, lang, requestedAttributes, tokenizer) {
 }
 
 /** Updates the HTML elements for a substring that will be added to the detailed analysis output */
-function addSubstring(substring, analysisContainer, result, loadingEl, response) {
+function addSubstring(substring, analysisContainer, result, loadingEl, response, lang) {
   // Check for errors and sort the attributes
-  if (typeof(response.error) != 'undefined') {
+  if (typeof(response.attributeScores) === 'undefined') {
     analysisContainer.removeChild(loadingEl);
     analysisContainer.appendChild(createAnyElement('p', 'Perspective API was not able to get scores for detailed analysis'));
     return;
   }
   const toxicityScore = response.attributeScores.TOXICITY.summaryScore.value;
   const attributes = sortAttributes(response);
 
-  // Color the substring appropriately	
-  const substringEl = createAnyElement('span', substring);
+  // Set up replacements function call and color the substring appropriately	
+  const substringEl = document.createElement('span');
+  const words = getSubstrings(substring, TokenizerEnum['WORD']);
+  for (let i = 0; i < words.length; i++) {
+    const wordEl = createAnyElement('span', words[i]);
+    const bareWord = words[i].replace(/[\s.,\/#!$%\^&\*;:{}=\-_`~()]/g, ''); // Remove any punctuation & whitespace
+    wordEl.onclick = function() { setUpReplacements(substring, lang, bareWord); }
+    wordEl.classList.add('output-word'); 
+    substringEl.appendChild(wordEl);
+  }
   colorSubstring(substringEl, toxicityScore);
      
   // Attach a tooltip (info-box for the substring)
   const tooltipEl = createTooltip(attributes);
   substringEl.appendChild(tooltipEl);
   result.appendChild(substringEl);
 }
+ 
+/** Sets up the replacements section */
+function setUpReplacements(text, lang, substringForReplacements) {
+  // Clear previous results
+  const headerContainer = document.getElementById('perspective-datamuse-header');
+  headerContainer.innerHTML = '';
+  document.getElementById('perspective-datamuse-chart').innerHTML = '';
+  document.getElementById('perspective-datamuse-extras').innerHTML = '';
+
+  // Set up about container 
```

Comment:
Just FYI, this code is fine but another options for hiding showing elements like this aboutContainer is to add them to the .html code with style `display: hidden`, then when you want to show them, change the style to `display: block` (or possibly inline or inline-block, depending on the specifics of your formatting).  You can also do this by adding and removing classes that set those same style properties.  IMO that style is a little easier to read, but it's up to you.

Reply:
Thanks, I didn't think of that! I changed the aboutContainer to this code style and I'll do it for other elements in future branches. Correct me if I'm wrong, but many of the elements in this section have dynamic output, so I believe this wouldn't apply to those.